pkg/restapi/restapi.go

```diff
@@ -40,3 +70,101 @@ type (
+
+	messageID, err := hornet.MessageIDFromHex(messageIDHex)
+	if err != nil {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid message ID: %s, error: %s", messageIDHex, err)
+	}
+	return messageID, nil
+}
+
+func ParseTransactionIDParam(c echo.Context) (*iotago.TransactionID, error) {
+	transactionIDHex := strings.ToLower(c.Param(ParameterTransactionID))
+
+	transactionIDBytes, err := hex.DecodeString(transactionIDHex)
+	if err != nil {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid transaction ID: %s, error: %s", transactionIDHex, err)
+	}
+
+	if len(transactionIDBytes) != iotago.TransactionIDLength {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid transaction ID: %s, invalid length: %d", transactionIDHex, len(transactionIDBytes))
+	}
+
+	var transactionID iotago.TransactionID
+	copy(transactionID[:], transactionIDBytes)
+	return &transactionID, nil
+}
+
+func ParseOutputIDParam(c echo.Context) (*iotago.UTXOInputID, error) {
+	outputIDParam := strings.ToLower(c.Param(ParameterOutputID))
+
+	outputIDBytes, err := hex.DecodeString(outputIDParam)
+	if err != nil {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid output ID: %s, error: %s", outputIDParam, err)
+	}
+
+	if len(outputIDBytes) != utxo.OutputIDLength {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid output ID: %s, error: %s", outputIDParam, err)
+	}
+
+	var outputID iotago.UTXOInputID
+	copy(outputID[:], outputIDBytes)
+	return &outputID, nil
+}
+
+func ParseBech32AddressParam(c echo.Context, prefix iotago.NetworkPrefix) (iotago.Address, error) {
+	addressParam := strings.ToLower(c.Param(ParameterAddress))
+
+	hrp, bech32Address, err := iotago.ParseBech32(addressParam)
+	if err != nil {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid address: %s, error: %s", addressParam, err)
+	}
+
+	if hrp != prefix {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid bech32 address, expected prefix: %s", prefix)
+	}
+
+	return bech32Address, nil
+}
+
+func ParseEd25519AddressParam(c echo.Context) (*iotago.Ed25519Address, error) {
+	addressParam := strings.ToLower(c.Param(ParameterAddress))
+
+	addressBytes, err := hex.DecodeString(addressParam)
+	if err != nil {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid address: %s, error: %s", addressParam, err)
+	}
+
+	if len(addressBytes) != (iotago.Ed25519AddressBytesLength) {
+		return nil, errors.WithMessagef(ErrInvalidParameter, "invalid address length: %s", addressParam)
+	}
+
+	var address iotago.Ed25519Address
+	copy(address[:], addressBytes)
+	return &address, nil
+}
+
+func ParseMilestoneIndexParam(c echo.Context) (milestone.Index, error) {
+	milestoneIndex := strings.ToLower(c.Param(ParameterMilestoneIndex))
+	if milestoneIndex == "" {
+		return 0, errors.WithMessagef(ErrInvalidParameter, "parameter \"%s\" not specified", ParameterMilestoneIndex)
+	}
+
+	msIndex, err := strconv.ParseUint(milestoneIndex, 10, 64)
```

Comment:
Guess that was my fault back then, but it should be
```suggestion
	msIndex, err := strconv.ParseUint(milestoneIndex, 10, 32)
```